### PR TITLE
[AIRFLOW-5062] Add support for getting/setting ACL in S3Hook

### PR DESCRIPTION
**Pull request by LesterKim**
### Jira

[AIRFLOW-5062](https://issues.apache.org/jira/browse/AIRFLOW-5062) Add ACL and get object permission

### Description

Adds ACL (access control list) header when uploading a file. Also, adds `get_object_permission` to see the S3 object's permission (given a bucket and key).

### Tests

My PR adds the following unit test: `test_get_object_permission`

### CODE REVIEW

**Comment by LesterKim** [comment]
#continuous-integration/travis-ci/pr

File: airflow/hooks/S3_hook.py [hunk 1]
```diff
@@ -318,7 +318,8 @@ def load_file(self,
                   key,
                   bucket_name=None,
                   replace=False,
-                  encrypt=False):
+                  encrypt=False,
+                  ACL=None):
```
Comment (potiuk) [inline airflow/hooks/S3_hook.py]: lowercase name will be still better even if it is an acronym

**Review by potiuk — CHANGES_REQUESTED** [review]
It would be nice to add some tests, especially that test_s3_hook already mocks S3 connection, it should be rather easy to add it - also including the missing tests for "encrypt" parameter.

File: airflow/hooks/S3_hook.py (hunk 1 above)
Comment (LesterKim) [reply airflow/hooks/S3_hook.py]: @potiuk Yes, I will replace `ACL` to `acl`.

**Comment by LesterKim** [comment]
@potiuk Yes, I noticed that some tests are missing. I can create a JIRA ticket and another pull request that adds tests for the S3 file upload methods.

**Comment by potiuk** [comment]
Tests should be part of this PR not a separate one. The reason is - if we decide to cherry-pick it to 1.10.x, we will not be sure if it still works there. Having a test covering it in single commit/PR is the only way we can safely cherry-pick such changes.

**Comment by mik-laj** [comment]
Travis is sad. Can you fix it?

**Comment by OmerJog** [comment]
@LesterKim  Can you please fix the error?

```
======================================================================
48) ERROR: test_load_file (tests.hooks.test_s3_hook.TestS3Hook)
----------------------------------------------------------------------
   Traceback (most recent call last):
    /usr/local/lib/python3.6/site-packages/moto/core/models.py line 71 in wrapper
      result = func(*args, **kwargs)
    tests/hooks/test_s3_hook.py line 267 in test_load_file
      hook.load_file(temp_file, "my_key", "mybucket")
    airflow/hooks/S3_hook.py line 56 in wrapper
      return func(*args, **kwargs)
    airflow/hooks/S3_hook.py line 432 in load_file
      client.upload_file(filename, bucket_name, key, ExtraArgs=extra_args)
    /usr/local/lib/python3.6/site-packages/boto3/s3/inject.py line 131 in upload_file
      extra_args=ExtraArgs, callback=Callback)
    /usr/local/lib/python3.6/site-packages/boto3/s3/transfer.py line 273 in upload_file
      raise ValueError('Filename must be a string')
   ValueError: Filename must be a string
   -------------------- >> begin captured logging << --------------------
   airflow.hooks.S3_hook.S3Hook: INFO: Not Found
   --------------------- >> end captured logging << ---------------------
```

**Comment by benjamingrenier** [comment]
Any update on this PR ?

**Pull request closed without merging**